src/main/javascript/crypto/src/IntegrationExample.ts

```diff
@@ -0,0 +1,203 @@
+
+            this.iframe.contentWindow.postMessage(sendData, this.attestationOrigin);
+            return;
+        }
+
+
+        if (
+            typeof event.data.display !== "undefined"
+        ) {
+            if (event.data.display === true) {
+                this.iframeWrap.style.display = 'flex';
+                this.negotiator && this.negotiator.commandDisplayIframe();
+            } else {
+
+                if (event.data.error){
+                    logger(DEBUGLEVEL.LOW, "Error received from the iframe: " + event.data.error);
+                    this.authResultCallback(null, event.data.error);
+                }
+
+                this.iframeWrap.style.display = 'none';
+                this.negotiator && this.negotiator.commandHideIframe();
+
+            }
+        }
+
+        if (
+            !event.data.hasOwnProperty('attestation')
+            || !event.data.hasOwnProperty('requestSecret')
+        ) {
+            return;
+        }
+        this.iframeWrap.remove();
+        this.attestationBlob = event.data.attestation;
+        this.attestationSecret = event.data.requestSecret;
+
+        logger(DEBUGLEVEL.HIGH,'attestation data received.');
+        logger(DEBUGLEVEL.HIGH,this.attestationBlob);
+        logger(DEBUGLEVEL.HIGH,this.attestationSecret);
+        logger(DEBUGLEVEL.HIGH,this.base64attestorPubKey);
+
+        try {
+            Authenticator.getUseTicket(
+                this.signedTokenSecret,
+                this.attestationSecret,
+                this.signedTokenBlob ,
+                this.attestationBlob ,
+                this.base64attestorPubKey,
+                this.base64senderPublicKey,
+            ).then(useToken => {
+                if (useToken){
+                    logger(DEBUGLEVEL.HIGH,'this.authResultCallback( useToken ): ');
+                    this.authResultCallback(useToken);
+                } else {
+                    logger(DEBUGLEVEL.HIGH,'this.authResultCallback( empty ): ');
+                    this.authResultCallback(useToken);
+                }
+
+            })
+
+
+        } catch (e){
+            logger(DEBUGLEVEL.LOW,`UseDevconTicket. Something went wrong. ${e}`);
+            this.authResultCallback(false);
+        }
+        // construct UseDevconTicket, see
+        // https://github.com/TokenScript/attestation/blob/main/data-modules/src/UseDevconTicket.asd
+
+        // TODO we dont have ready UseDevconTicket constructor yet
+        // let useDevconTicket = new UseDevconTicket({
+        //     signedDevconTicket: signedDevonTicket,
+        //     identifierAttestation: identifierAttestation,
+        //     proof: proof
+        // })
+        // // Serialise it (for use as a transaction parameter) and return it
+        // return useDevconTicket.serialize();
+    }
+
+    attachPostMessageListener(listener: Function){
+        if (window.addEventListener) {
+            window.addEventListener("message", (e) => {
```

Comment:
To avoid potential cross origin attacks, it should be validated that the origin of the message is as expected, ie. something like the following should be added
``` 
if (event.origin !== "http://example.org") // Compliant
    return;
```

Reply:
this part of code just add event listener , so we have to check event.origin in the callback function. this part of code doesnt have access to the event object